portal.go

```diff
@@ -703,8 +703,17 @@ func (portal *Portal) encryptFile(data []byte, mimeType string) ([]byte, string,
 	return file.Encrypt(data), "application/octet-stream", file
 }
 
-func (portal *Portal) sendErrorMessage(evt *event.Event, err error, isCertain bool) id.EventID {
-	go portal.bridge.AS.SendErrorMessageSendCheckpoint(evt, appservice.StepRemote, err, true, 0)
+func (portal *Portal) sendErrorMessage(evt *event.Event, err error, isCertain bool, unsupported bool) id.EventID {
+	go (func() {
+		status := appservice.StatusPermFailure
+		if unsupported {
+			status = appservice.StatusUnsupported
+		}
+		checkpoint := appservice.NewMessageSendCheckpoint(evt, appservice.StepRemote, status, 0)
+		checkpoint.Info = err.Error()
+		checkpoint.Send(portal.bridge.AS)
+	})()
```

Comment:
I don't think the checkpoint preparation needs to be in the goroutine

```suggestion
	status := appservice.StatusPermFailure
	if unsupported {
		status = appservice.StatusUnsupported
	}
	checkpoint := appservice.NewMessageSendCheckpoint(evt, appservice.StepRemote, status, 0)
	checkpoint.Info = err.Error()
	go checkpoint.Send(portal.bridge.AS)
```